File: core/src/commonMain/kotlin/work/socialhub/kmastodon/Page.java
package work.socialhub.kmastodon;

import java.util.Optional;

public final class Page {

    private Optional<Long> offset;
    private Optional<Long> limit;

    public Page() {
        this.offset = Optional.empty();
        this.limit = Optional.empty();
    }

    public Optional<Long> getOffset() {
        return offset;
    }

    public void setOffset(long offset) {
        this.offset = Optional.of(offset);
    }

    public Optional<Long> getLimit() {
        return limit;
    }

    public void setLimit(long limit) {
        this.limit = Optional.of(limit);
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kmastodon/Page.kt
package work.socialhub.kmastodon

class Page {
    private var offset: java.util.Optional<Long>
    private var limit: java.util.Optional<Long>

    init {
        this.offset = java.util.Optional.empty<Long>()
        this.limit = java.util.Optional.empty<Long>()
    }

    fun getOffset(): java.util.Optional<Long> {
        return offset
    }

    fun setOffset(offset: Long) {
        this.offset = java.util.Optional.of<Long>(offset)
    }

    fun getLimit(): java.util.Optional<Long> {
        return limit
    }

    fun setLimit(limit: Long) {
        this.limit = java.util.Optional.of<Long>(limit)
    }
}
